service/routes.py

```diff
@@ -81,57 +81,55 @@ def create_wishlists():
     ),
     status.HTTP_201_CREATED
   )
-
+  
 ######################################################################
 # List ALL WISHLISTS
 ######################################################################
 @app.route("/wishlists", methods=["GET"])
 def list_all_wishlists():
   """
-  Lists all wishlists
+  Lists all wishlists 
   This endpoint will return all wishlists in the database.
-  """
-  app.logger.info("Request for all wishlists")
-  query_res = Wishlist.find_all()
-
-  wishlists = [r for r in query_res]
-  res = []
-  for wishlist in wishlists:
-    wishlist_products = Product.find_all_by_wishlist_id(wishlist.id)
-    res.append(WishlistVo(wishlist,wishlist_products))
 
-  result_ser = [vo.serialize() for vo in res]
+  Or list wishlists by user_id
+  """
+  user_id = request.args.get('user_id')
+  if not user_id:
+    app.logger.info("Request for all wishlists")
+    query_res = Wishlist.find_all()
+
+    wishlists = [r for r in query_res]
+    res = []
+    for wishlist in wishlists:
+      wishlist_products = Product.find_all_by_wishlist_id(wishlist.id)
+      res.append(WishlistVo(wishlist,wishlist_products))
+
+    result_ser = [vo.serialize() for vo in res]
+
+    if not result_ser:
```

Comment:
I know that this was like this before, but I think the professor said that a LIST function should ever return a 404. If `wishlists ` is an empty array, we can just return it as `{data: [], message: 'No wishlists found!'}` after line 101. If not - do all the serialization. Could you rearrange this logic please?